src/lib.rs

```diff
@@ -94,7 +95,99 @@ fn serde_to_starlark<'v>(x: serde_json::Value, heap: &'v Heap) -> anyhow::Result
 
 // }}}
 
+// {{{ PythonObjectValue
+
+#[derive(Debug, ProvidesStaticType, NoSerialize, Allocative)]
+struct PythonObjectValue {
+    #[allocative(skip)]
+    obj: PyObject,
+}
+starlark_simple_value!(PythonObjectValue);
+
+impl Display for PythonObjectValue {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        Python::with_gil(|py| write!(f, "<python object: {}>", self.obj.bind(py).repr().unwrap()))
+    }
+}
+
+#[starlark_value(type = "python_object")]
+impl<'v> StarlarkValue<'v> for PythonObjectValue {
+    fn invoke(
+        &self,
+        _me: Value<'v>,
+        args: &Arguments<'v, '_>,
+        eval: &mut starlark::eval::Evaluator<'v, '_>,
+    ) -> starlark::Result<Value<'v>> {
+        Python::with_gil(|py| -> starlark::Result<Value<'v>> {
+            // Handle positional arguments
+            let py_args: Vec<PyObject> = convert_to_starlark_err(
+                (args
+                    .positions(eval.heap())?
+                    .map(|v| -> PyResult<PyObject> { value_to_pyobject(v) }))
+                .collect::<PyResult<Vec<PyObject>>>(),
+            )?;
+
+            // Handle named arguments
+            let py_kwargs = PyDict::new_bound(py);
+            for name in args.names_map()?.iter() {
+                let key = name.0.as_str();
+                let val = convert_to_starlark_err(value_to_pyobject(*name.1))?;
+                convert_to_starlark_err(py_kwargs.set_item(key, val))?;
+            }
```

Comment:
The tuple could be unpacked directly
```suggestion
            for (key, value) in args.names_map()?.iter() {
                convert_to_starlark_err(py_kwargs.set_item(
                    key.as_str(),
                    convert_to_starlark_err(value_to_pyobject(*value))?,
                ))?;
            }
```